src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -63,15 +64,29 @@ public Page<AdvertisementTitleDto> findPaginatedAsThumbnails(
             @Parameter(name = "Results page you want to retrieve (0..N). Default value: 0")
             @RequestParam(value = "page", required = false, defaultValue = "0") @PositiveOrZero int page,
             @Parameter(name = "Number of records per page. Default value: 12")
-            @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size) {
+            @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size,
+            @Parameter(name = "ID of existed advertisement")
+            @RequestParam(value = "advertisementId", required = false) UUID advertisementId,
+            @Parameter(name = "ID of existed subcategory")
+            @RequestParam(value = "subcategoryId", required = false) Long subcategoryId) {
+        if (advertisementId != null && subcategoryId != null) {
+            return advertisementService.findAllThumbnails(PageRequest.of(page, size), advertisementId, subcategoryId);
+        }
+
         return advertisementService.findAllThumbnails(PageRequest.of(page, size));
```

Comment:
You should do all the search with a single method (your new method) and the current one has to be removed.